src/ngraph/pass/algebraic_simplification.cpp

```diff
@@ -543,6 +544,73 @@ static bool simplify_reduction(shared_ptr<Node> n)
     return true;
 }
 
+static bool replace_transpose_with_reshape(shared_ptr<Node> n)
+{
+    bool rc = false;
+    auto transpose = as_type_ptr<op::Transpose>(n);
+
+    if (transpose)
+    {
+        auto perm = as_type_ptr<op::Constant>(n->get_argument(1));
+        auto perm_value = perm->get_vector<int64_t>();
+        auto data = n->get_argument(0);
+        auto data_shape = n->input_value(0).get_partial_shape().to_shape();
```

Comment:
You should check that the rank is static. Then you should iterate over the partial shape which may have dynamic dimensions. The dynamic dimensions should not be a blocker to perform this optimization. If the dimension is dynamic then we should consider it as not equal to 1.

Reply:
After some time of reviewing the code it seems that the dynamic input shape may be a problem. If there is just one dynamic dimension then we can insert value "-1" for this dimension in the Reshape second input, but if there are more than two dynamic dimensions then we cannot perform this transformation. We need to cover this cases in unit-tests.

Reply:
But the general solution will be the following: Insert ShapeOf operation which gets input from the input to Transpose. Then pass output from ShapeOf to operation Gather as first input. The second input to Gather will be the permute_order from Transpose. So the result of Gather operation should go as second input to Reshape operation.

Reply:
It supports 1 unknown dim